kedro/io/data_catalog.py

```diff
@@ -311,6 +428,28 @@ def _get_dataset(
 
         return data_set
 
+    def _resolve_config(self, data_set_name, data_set_pattern) -> dict[str, Any]:
+        """Get resolved config from a patterned config with placeholders
+        replaced with actual values"""
+        result = parse(data_set_pattern, data_set_name)
+        config_copy = self._raw_catalog[data_set_pattern].copy()
+        if data_set_name in self._raw_catalog:
+            # Merge config with entry created containing load and save versions
+            config_copy.update(self._raw_catalog[data_set_name])
+        for key, value in config_copy.items():
+            if isinstance(value, Iterable) and "}" in value:
```

Comment:
Why do we check for `"{"` sometimes but `"}"`? Again can we just use one function to check if something is a pattern?